src/base/base.service.ts

```diff
@@ -0,0 +1,15 @@
+import { BaseEntity, Repository } from 'typeorm';
+
+export default class BaseService<Entity extends BaseEntity> {
+  constructor(
+    public repository: Repository<Entity>,
+  ) {}
+
+  async findById(id: number): Promise<Entity> {
+    return await this.repository.findOne(id);
+  }
+
+  async findAll(): Promise<Entity[]> {
+    return await this.repository.find();
+  }
```

Comment:
Why do we need async/await here if we just return Promise?

Reply:
Fixed